### DISCUSSION TITLE
flutter_flavorizr with FlutterFire CLI

### DISCUSSION THREAD

**Discussion by mavinis** [AngeloAvv/flutter_flavorizr#105] [General]
Any plans on making flavorizr work with FlutterFifre CLI?

FlutterFire makes Firebase configuration a lot easier, but it no longer generate a `GoogleService-Info.plist` file. It creates a `firebase_app_id_file.json` instead. It would be awesome if flavorizr could add Firebase configurations for this file for each flavor too.

**Comment by paulinventome** [comment]
I am running flavorizr and flutterfire. But I need to find a way to have different app_id and plist files for each flavour. The code inside flutter is easier because I can use different options files for each flavour but at build time the two files that get replaced are GoogleService-Info.plist (which seems to be needed for android) and firebase_app_id_file.json

So I think it is possible somehow now to have different files used for each flavour but I don't know how. However it would also be awesome is this was built in. AFAIK it is literally those two files that would need swapping each time

**Reply by paulinventome** [reply]
Sorry my mistake, android is google-services.json not plist (doh)

**Comment by AngeloAvv** [comment]
The simpliest answer would be no, because it means bringing all the code from flutterfire CLI into flutter_flavorizr, turning it into a duplicate of the existing CLI.

flutter_flavorizr works out of the box using google-services.json and GoogleService-Info.plist, the only drawback is that you need to define gradle dependencies manually.

You don't need the flutterfireCLI autogenerated files to let your Flutter project work with Firebase.

**Comment by paulinventome** [comment]
Okay but am I wrong in believing superficially all I need for this to work is to get the build process to use/copy the correct file based on flavor/scheme?

In Android this is supposed to happen for correctly named folders and in iOS I could add an Xcode build script that copied the correct file.

Or is it actually more complicated than that?

Kindest
Paul

**Reply by AngeloAvv** [reply]
You don't need to copy anything, everything is in charge of flutter_flavorizr, it copies the google-services.json in the right folders for both Android and iOS, and for iOS it also adds a Build Script to switch the GoogleService-Info.plist automatically across the different flavors builds.
